.github/workflows/lint_test_build.yml

```diff
@@ -102,6 +102,27 @@ jobs:
         run: |
           npx @stoplight/spectral-cli lint ./packages/token-introspection/openapi/*.yaml
 
+  integration-test:
+    runs-on: ubuntu-22.04
+    needs: checkout
+    timeout-minutes: 5
```

Comment:
Originally this was a seperate workflow file but I figured I'd include it here since it's the same event trigger.

I considered making this depend on auth, backend, token-introspection but decided not to because I think it gives us a useful signal if integration tests pass and auth/backend do not. This could happen if the failures just dont overlap with the tests cases, or if the backend/auth tests are flakey. Additionally, not making it depend on our longest running jobs ensures that we aren't increasing the time it takes to run CI. The integration tests take ~2 minutes, so they should finish before backend.